File: service/go/model/code.go
package model

type Code struct {
	Type     string      `bson:"type,omitempty" json:"type,omitempty"`
	Market   int         `bson:"market,omitempty" json:"market,omitempty"`
	Zsz      interface{} `bson:"zsz,omitempty" json:"zsz,omitempty"`
	Ltgb     interface{} `bson:"ltgb,omitempty" json:"ltgb,omitempty"`
	Code     string      `bson:"code,omitempty" json:"code,omitempty"`
	Zgb      interface{} `bson:"zgb,omitempty" json:"zgb,omitempty"`
	Name     string      `bson:"name,omitempty" json:"name,omitempty"`
	Bk       string      `bson:"bk,omitempty" json:"bk,omitempty"`
	Roe      float64     `bson:"roe,omitempty" json:"roe,omitempty"`
	Ltsz     interface{} `bson:"ltsz,omitempty" json:"ltsz,omitempty"`
	Ssdate   string      `bson:"ssdate,omitempty" json:"ssdate,omitempty"`
	UpdateAt int64       `bson:"update_at,omitempty" json:"update_at,omitempty"`
}

File: service/rust/src/model/code.rs
use anyhow::{Context, Result};
use mongodb::{bson::doc, results, Collection};
use serde::{Deserialize, Serialize};
#[derive(Debug, Serialize, Deserialize, Default)]
pub struct Code {
    pub type_: String,
    pub data: SourceCode,
}

#[derive(Debug, Serialize, Deserialize, Default, Clone)]
pub struct SourceCode {
    pub code: String,
    pub name: String,
    pub market: i32,
    pub hsgt: i32,
    pub bk: String,
    pub roe: f32,
    pub zgb: f64,
    pub ltgb: f64,
    pub ltsz: f64,
    pub zsz: f64,
    pub ssdate: String,
}

impl SourceCode {
    pub fn to_code(self, type_: String) -> Code {
        Code {
            type_,
            data: self.into(),
        }
    }
}

pub async fn insert_many(l: Vec<Code>, c: Collection<Code>) -> Result<results::InsertManyResult> {
    c.insert_many(l, None).await.context("code insert_many err")
}
// doc! { "author": "George Orwell" }

pub async fn insert(i: Code, c: Collection<Code>) -> Result<results::InsertOneResult> {
    c.insert_one(i, None).await.context("code insert err")
}

impl Code {
    pub fn to_stock(&mut self) {
        self.type_ = "stock".to_string();
    }
    pub fn to_fund(&mut self) {
        self.type_ = "fund".to_string();
    }
    pub fn is_stock(&self) -> bool {
        self.type_ == "stock".to_string()
    }
    pub fn is_fund(&self) -> bool {
        self.type_ == "fund".to_string()
    }
}
